inc/Engine/Optimization/Minify/JS/AbstractJSOptimization.php

```diff
@@ -94,12 +104,13 @@ protected function is_minify_excluded_file( array $tag ) {
 		if ( ! isset( $tag[0], $tag['url'] ) ) {
 			return true;
 		}
+		$exclude_js_template = $this->dynamic_lists->get_exclude_js_templates();
 
 		// File should not be minified.
 		if (
-			false !== strpos( $tag[0], 'data-minify=' )
+			in_array( 'data-minify=', $exclude_js_template, true )
 			||
-			false !== strpos( $tag[0], 'data-no-minify=' )
+			in_array( 'data-no-minify=', $exclude_js_template, true )
```

Comment:
I believe we should change this part completely to be regex to compare the current tag `$tag[0]` with the values from the dynamic list, we don't need to check specific values here like `data-no-minify=` we need to make it more generic.

Reply:
spot on, the list might grow and we don't have to keep modifying the code for that